### DISCUSSION TITLE
Install packages once, use them in all projects

### DISCUSSION THREAD

**Discussion by erfanasbari** [npm/feedback#160] [General]
It would be really good if there was an option to install packages once and use them in multiple projects.

Because right now every time I want to create a project ( For example create-react-app ) I have to wait for NPM to download over 200Mg of packages. and because of huge amount of items ( over 30,000 files ) it's really hard to copy theme or do anything to them!

But if there was a way to just install them once, all of these problem would not exist!

**Comment by MylesBorins** [comment]
Hey @EInterStellar

It is possible to set up a global folder and set [a NODE_PATH](https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders) if you want to only install once. That said I don't really advise this approach as it significantly hurts reusability if you don't track things in the package.json and a local node_modules folder.

Another option is to run `npm install --prefer-offline` which will default to using locally cached assets instead of using the network. I've been recently working using a very low-speed satellite internet and this has been a massive help.

As the resolution of modules is handled by node, and there is a way to get node to load from arbitrary folders as mentioned above, I don't think there is anything actionable for the npm team today. Please feel free to ask additional questions if you have any.

**Reply by erfanasbari** [reply]
Thanks a lot for responding.

I found a solution as well. just by installing modules in a parent directory.

But it would be cool if the amount of files were less, for example ~1,000 files for around 8 packages instead of ~30,000 files.
I don't know this amount of files is because of package itself or NPM.

For example I have `node_modules` folder that contains 10 - 15 packages and it size is 210Mg but size on disk is 250Mg because of each file info and indexes. And when I try to transfer this folder or compress it or any other operation it takes for ever to process.

I know it's too much but it would be really good if we could install compiled modules in future so it takes a lot less files and folders.

Anyways thanks for your support. I Really appreciate it. <3
